tests/baselines/reference/arrayFilterBooleanOverload#56013WithoutExternalOverload.types

```diff
@@ -19,31 +19,31 @@ const id = <T,>() => (t: T) => !!t;
 
 ['foo', 'bar'].filter(id()); // // expect id() = (t: string) => boolean
 >['foo', 'bar'].filter(id()) : string[]
->['foo', 'bar'].filter : { <S extends string>(predicate: (value: string, index: number, array: string[]) => value is S, thisArg?: any): S[]; (predicate: (value: string, index: number, array: string[]) => unknown, thisArg?: any): string[]; }
+>['foo', 'bar'].filter : { <S extends string>(predicate: (value: string, index: number, array: string[]) => value is S, thisArg?: any): S[]; (predicate: BooleanConstructor, thisArg?: any): string[]; (predicate: (value: string, index: number, array: string[]) => unknown, thisArg?: any): string[]; }
 >['foo', 'bar'] : string[]
 >'foo' : "foo"
 >'bar' : "bar"
->filter : { <S extends string>(predicate: (value: string, index: number, array: string[]) => value is S, thisArg?: any): S[]; (predicate: (value: string, index: number, array: string[]) => unknown, thisArg?: any): string[]; }
+>filter : { <S extends string>(predicate: (value: string, index: number, array: string[]) => value is S, thisArg?: any): S[]; (predicate: BooleanConstructor, thisArg?: any): string[]; (predicate: (value: string, index: number, array: string[]) => unknown, thisArg?: any): string[]; }
 >id() : (t: string) => boolean

```

Comment:
In #56013 (bug), `id()` was not evaluated when an external `filter(Boolean)` overload was added to the Array interface  in usercode.  Now it works (with the overload in TypeScript code).